### PR TITLE
Improve code based on tool analysis feeback

### PR DESCRIPTION
**Pull request by mikegray**
Fixes are based on Go report card output:
- `gofmt -s` eliminates some duplication in struct/slice initialization
- `golint` found some issues like: `warning: should drop = nil from declaration of var XXX; it is the zero value`
- `golint` found some issues like: `warning: receiver name XXX should be consistent with previous receiver name YYY for ZZZ`
- `golint` says not to use underscores for function/variable names
- `golint` found several issues like: `warning: if block ends with a return statement, so drop this else and outdent its block`

No functional changes are included - just source code quality improvements.

### CODE REVIEW

**Comment by mikegray** [comment]
@willhite PTAL

File: go/types/struct.go [hunk 1]
```diff
@@ -126,19 +126,19 @@ func (s Struct) Set(n string, v Value) Struct {
 	return Struct{values, s.t, &hash.Hash{}}
 }
 
-func (s1 Struct) Diff(s2 Struct, changes chan<- ValueChanged, closeChan <-chan struct{}) {
-	if s1.Equals(s2) {
+func (s Struct) Diff(last Struct, changes chan<- ValueChanged, closeChan <-chan struct{}) {
+	if s.Equals(last) {
```
Comment (willhite) [inline go/types/struct.go]: Just a suggestion, would "other" be a better variable name than "last"?

Comment (mikegray) [inline go/types/struct.go]: map, list, set all use last.

**Comment by willhite** [comment]
Nice. LGTM

**Pull request merged**